src/libraries/Common/src/Interop/Unix/System.Native/Interop.MountPoints.FormatInfo.cs

```diff
@@ -50,8 +50,68 @@ internal static int GetFormatInfoForMountPoint(string name, out DriveType type)
             return GetFormatInfoForMountPoint(name, out _, out type);
         }
 
+        /// <summary>
+        /// Retrieves format information for the specified mount point.
+        /// </summary>
+        /// <remarks>
+        /// This method uses a two-step approach to retrieve filesystem information:
+        /// 1. On Linux systems, it first attempts to read from `/proc/self/mountinfo`.
+        /// 2. If step 1 fails or on non-Linux systems, it falls back to a P/Invoke call.
+        ///
+        /// The `/proc/self/mountinfo` approach is preferred on Linux because
+        /// it's more reliable when procfs is available and functioning correctly.
+        ///
+        /// For other systems:
+        /// - SunOS and similar: procfs provides a binary interface, not suitable for this method.
+        /// - macOS: procfs is not available.
+        /// - FreeBSD: procfs is optional and not enabled by default.
+        ///
+        /// The method uses try/catch blocks to ensure robustness, even though `/proc/self/mountinfo`
+        /// should be specific to each process according to kernel documentation.
+        /// </remarks>
+        /// <param name="name">The mount point name to query.</param>
+        /// <param name="format">Output parameter for the filesystem format.</param>
+        /// <param name="type">Output parameter for the drive type.</param>
+        /// <returns>0 if successful, otherwise an error code.</returns>
         private static unsafe int GetFormatInfoForMountPoint(string name, out string format, out DriveType type)
         {
+#if TARGET_LINUX
+            try
+            {
+                const string mountInfoFilePath = "/proc/self/mountinfo";
+                var mountInfoFileContent = File.ReadAllLines(mountInfoFilePath);
+                foreach (var line in mountInfoFileContent)
+                {
+                    var parser = new StringParser(line, ' ');
+            
+                    // Skip fields we don't care about (Fields 1-4)
+                    parser.MoveNext(); // Skip Mount ID
+                    parser.MoveNext(); // Skip Parent ID
+                    parser.MoveNext(); // Skip Major:Minor
+                    parser.MoveNext(); // Skip Root
+            
+                    // Get the mount point (Field 5)
+                    string mountPoint = parser.MoveAndExtractNext();
+            
+                    // Skip to the separator which is end of optional fields (Field 8)
+                    while (parser.MoveAndExtractNext() != "-")
+                    {
+                    }
+            
+                    // Get filesystem type (Field 9)
+                    string filesystemType = parser.MoveAndExtractNext();
+            
+                    if (mountPoint.Equals(name, StringComparison.Ordinal))
```

Comment:
```suggestion
                    if (mountPoint == name)
```
Do not use the instance `Equals` method. `==` uses `Ordinal` behavior.